### PR TITLE
InterestAccrual: Unify current & previous debt methods

### LINKED ISSUE #1203: InterestAccrual: Unify previous_debt() and current_debt() methods
### Which part of the code is the issue addressing?

* Refactoring

### Description

`current_debt()` can be seen as a `previous_debt()` for `when` == `now()`. Using only one method helps to reason about "getting the debt" an InterestAccrual user.

### Research/based on

Slack [thread](https://kflabs.slack.com/archives/C04HJN404RK/p1675667938325789)

### How will this affect the code base

Helps to reason about the `InterestAccrual` trait.

### What are forseen obstacles or hurdles to overcome?

Currently, `current_debt()` is a separated method because it's faster to get the current than the previous debt. Merging both actions requires checking this difference in order not to lose performance.

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

Fixes #1203 

## Changes and Descriptions

- `current_debt` and `previous_debt` merged into `calculate_debt`.

### CODE REVIEW

File: pallets/interest-accrual/src/lib.rs [hunk 1]
```diff
@@ -335,30 +335,35 @@ pub mod pallet {
...
-			Self::calculate_debt(normalized_debt, past_rate)
+
+			let acc_rate = match when.cmp(&now) {
+				Ordering::Equal => rate.accumulated_rate,
+				Ordering::Less => {
+					let delta = now.ensure_sub(when)?;
+					let rate_adjustment =
+						checked_pow(rate.interest_rate_per_sec, delta.ensure_into()?)
+							.ok_or(ArithmeticError::Overflow)?;
+					rate.accumulated_rate.ensure_div(rate_adjustment)?
```
Comment (mustermeiszer) [inline pallets/interest-accrual/src/lib.rs:361]: Minor: but I like 

`Match when.cmp(now)`

Makes reading easier IMO and control flow safer.

**Review by mustermeiszer — APPROVED** [review]
Just a nit-pick. Looks good!

**Comment by mustermeiszer** [comment]
Haha. Just changed when I wrote this.

File: pallets/interest-accrual/src/lib.rs (hunk 1 above)
Comment (lemunozm) [reply pallets/interest-accrual/src/lib.rs:361]: Yeah, I agree 🤣

**Pull request merged**